File: Guanabara - POO Java/Meus exercícios POO/Aula12.Polimorfismo/src/main/java/polimorfismo/Reptil.java
package polimorfismo;
public class Reptil extends Animal {//tipo de herança: 'para diferença'
    //Atributo
    private String corEscama;
    
    //Implementando Métodos Abstratos (Polimorfismo de Sobreposição)
    @Override
    public void locomover() {
        System.out.println("Rastejando...");
    }

    @Override
    public void alimentar() {
        System.out.println("Comendo vegetais...");
    }

    @Override
    public void emitirSom() {
        System.out.println("Som de réptil...");
    }
    
    //Métodos Acessores
    public String getCorEscama() {
        return corEscama;
    }

    public void setCorEscama(String corEscama) {
        this.corEscama = corEscama;
    }
    
}

File: Guanabara - PHP/exercicios-PHPOO/Aula12-Polimorfismo/Reptil.php
<?php
include_once 'Animal.php';
class Reptil extends Animal
{
    //Atributo
    protected $corEscama;
    
    //Métodos Implementados usando polimorfismo de sobreposição
    public function alimentar() {
        echo '<p>Réptil se alimentando...</p>';
    }
    public function emitirSom() {
        echo '<p>Som de Réptil...</p>';
    }
    public function locomover() {
        echo '<p>Réptil se arrastando...</p>';
    }
    
    //Métodos Acessores
    public function getCorEscama() {
        return $this->corEscama;
    }
    public function setCorEscama($corEscama): void {
        $this->corEscama = $corEscama;
    }
}
